### PR TITLE
Add CUE module for generating Kubernetes manifests 

### PR DESCRIPTION
**Pull request by phoban01**
This PR adds a CUE module to generate a `podinfo` application resource set.

It defines a package `podinfo` which contains definitions for the objects declared in the podinfo Helm Chart. An instance of `podinfo` can be generated by createing a new `podinfo.#Application` and passing `podinfo.#Config`.

In addition, a cue tool command has been provided to easily generate yaml manifests:
```shell
cue gen
```

Signed-off-by: Piaras Hoban <phoban01@gmail.com>

### CODE REVIEW

File: cuelang/podinfo/config.cue [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+#Config: {
+	meta: metav1.#ObjectMeta
+	image: {
+		repository: *"ghcr.io/stefanprodan/podinfo" | string
+		tag:        "6.1.1" | string
```
Comment (stefanprodan) [inline cuelang/podinfo/config.cue:12]: Please remove the default from here, a tag should be required.

File: cuelang/main.cue [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	input: {
+		meta: {
+			name: "podinfo"
+			annotations: {
+				"app.kubernetes.io/name": "podinfo"
+			}
+		}
+		image: {
+			repository: "ghcr.io/stefanprodan/podinfo"
+			tag:        "6.0.3"
```
Comment (stefanprodan) [inline cuelang/main.cue:17]: Please extent `make version-set` to patch the tag.

File: cuelang/podinfo/linkerd.cue [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package podinfo
+
+#ServiceProfile: {
```
Comment (stefanprodan) [inline cuelang/podinfo/linkerd.cue:3]: Let's remove this for now, the profile could be generated from podinfo API spec using CUE, but we can do this in another PR.

**Review by stefanprodan — COMMENTED** [review]
@phoban01 can you please add cue vet and fmt to the test.yaml workflow? Something like:

```yaml
      - name: Setup CUE
        uses: cue-lang/setup-cue@main
      - name: Test
        run: cue vet ./cuelang/podinfo
```

Also please rename the dir from `cuelang` to just `cue`. Thanks!

**Comment by phoban01** [comment]
Thanks for the review @stefanprodan; have addressed these issues now.
